src/pages/company-info/index.js

```diff
@@ -0,0 +1,188 @@
+import { Box, Grid } from '@mui/material'
+import React, { useContext, useEffect, useRef, useState } from 'react'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import FormShell from 'src/components/Shared/FormShell'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import useResourceParams from 'src/hooks/useResourceParams'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import toast from 'react-hot-toast'
+import { useForm } from 'src/hooks/form'
+import ImageUpload from 'src/components/Inputs/ImageUpload'
+import { getStorageData } from 'src/storage/storage'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+
+const CompanyInfo = () => {
+  const imageUploadRef = useRef()
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const { labels: labels, access: maxAccess } = useResourceParams({
+    datasetId: ResourceIds.CompanyInfo
+  })
+
+  useEffect(() => {
+    getData()
+  }, [])
+
+  async function getData() {
+    const res = await getRequest({
+      extension: SystemRepository.CompanyInfo.get,
+      parameters: `_filter=`
+    })
+    res.record.accountId = getStorageData('userData')?.accountId
+    formik.setValues({
+      name: res.record.name,
+      webSite: res.record.taxNo,
+      taxNo: res.record.taxNo,
+      licenseNo: res.record.licenseNo,
+      crNo: res.record.crNo,
+      logoUrl: res.record.logoUrl,
+      flName: res.record.flName
+    })
+  }
+
+  const { formik } = useForm({
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues: {
+      plantId: '',
+      accountId: getStorageData('userData')?.accountId,
+      name: '',
+      webSite: '',
+      taxNo: '',
+      licenseNo: '',
+      crNo: '',
+      logoUrl: '',
+      flName: ''
+    },
+    onSubmit: values => {
+      post(values)
+    }
+  })
+
+  const post = async obj => {
+    const result = await postRequest({
+      extension: SystemRepository.CompanyInfo.set,
+      record: JSON.stringify({ ...obj, attachment: null })
+    })
+    if (imageUploadRef.current) {
+      imageUploadRef.current.submit()
```

Comment:
await on this too. Make sure that imageUploadRef.current.submit returns a promise (is an async function)